File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.js
function check_want(market,want,number)
{
    for (let k = 0; k < want.length; k++)
        {
            if(number[k] > market.reduce((count, data) => 
             data == want[k] ? count + 1 : count, 0))
                {
                    return 0;
                }
        }
    return 1;
}

function solution(want, number, discount) {
    var answer = 0;
    let market = [];
    for (let i = 0 ; i <= discount.length-10; i++)
        {
            for (let k = i; k<discount.length; k++)
                {
                    if(market.length == 10) break;
                    market.push(discount[k]);
                }
            answer+=check_want(market,want,number);
            market.length = 0;
        }
    return answer;
}

File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int check_want(vector<string> v,vector<string> want, vector<int> number)
{
    for (int i = 0 ; i < want.size(); i++)
    {
      if(count(v.begin(), v.end(), want[i]) < number[i])
      {
          return 0;
      }
    }
    return 1;
}
int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    vector<string> market;
    
    for (int i = 0 ; i <=discount.size()-10; i++)
    {
        for(int k = i; k < discount.size(); k++)
        {
            if (market.size() == 10) break;
            market.push_back(discount[k]);
        }
        answer+= check_want(market,want,number);
        market.clear();
        
    }
    return answer;
}